### PR TITLE
Jenkins: Add Mattermost deployment notifications

### PR DESCRIPTION
**Pull request by OrlandoSoto**
This PR notifies via mattermost the result of the build or deploy so that it is not required to look at the Jenkins log to obtain that information

## Additions

- Add mattermost notification at the end of each pipeline run
- Add mattermost notification on a successful PR deployment
- Notification posts to `foobar` though this can be configured for any more relevant mattermost channel

## Removals

- None

## Changes

- None

## Testing

1. A mattermost notification is sent to the foobar channel when the pipeline runs and when the deploy is successful

## Notes

- The deployment notification includes a link to the PR, Change Author, Jenkins build URL and stack URL

- There is also a notification that is always generated and includes the Pipeline build result, Jenkins job name and Git Branch Name

## Todos

-

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -107,4 +108,10 @@ pipeline {
             }
         }
     }
+
+    post {
+        always {
+            notify("${NOTIFICATION_CHANNEL}", "${currentBuild.result} ${env.JOB_NAME} ${env.BRANCH_NAME}")
```
Comment (hkeeler) [inline Jenkinsfile:114]: I think this should be a richer message, with links to as many useful resources as possible.  Maybe something like:

>@hkeeler deployed PR #5647 via Jenkins job [cfgov-refresh » PR-5647 » #4]()
>
>View site at: https://cfgov-pr-5647...
>View stack at: https://docker-cluster/stacks/cfgov-pr-5647/...

I also think you might want to spit up the `always` into `success` and `unsuccessful` conditions, as I think the messages will likely be pretty different.

Also also, we may want to incorporate this into the shared library itself.  I'm guessing the patterns will be pretty similar across apps, and it'd cut down on a fair bit of boilerplate if we did that.

Comment (OrlandoSoto) [reply Jenkinsfile:115]: @hkeeler how would you get the url to the PR inside the Pipeline?

File: Jenkinsfile [hunk 2]
```diff
@@ -104,7 +105,14 @@ pipeline {
                     dockerStack.deploy(env.STACK_NAME, 'docker-stack.yml')
                 }
                 echo "Site available at: https://${CFGOV_HOSTNAME}"
+                notify("${NOTIFICATION_CHANNEL}", "PR ${env.CHANGE_URL} deployed by ${env.CHANGE_AUTHOR} via ${env.BUILD_URL} and available at: https://${CFGOV_HOSTNAME}")
```
Comment (Scotchester) [inline Jenkinsfile:108]:
```suggestion
                notify("${NOTIFICATION_CHANNEL}", ":white_check_mark: PR ${env.CHANGE_URL} deployed by ${env.CHANGE_AUTHOR} via ${env.BUILD_URL} and available at https://${CFGOV_HOSTNAME}.")
```

File: Jenkinsfile [hunk 3]
```diff
@@ -104,7 +105,14 @@ pipeline {
...
             }
         }
     }
+
+    post {
+        always {
+            notify("${NOTIFICATION_CHANNEL}", "${currentBuild.result} ${env.JOB_NAME} ${env.BRANCH_NAME}")
```
Comment (Scotchester) [inline Jenkinsfile:115]:
```suggestion
        unsuccessful {
            notify("${NOTIFICATION_CHANNEL}", ":x: PR ${env.CHANGE_URL} by ${env.CHANGE_AUTHOR} failed to deploy. See: ${env.BUILD_URL}.")
```

**Comment by OrlandoSoto** [comment]
Thanks @Scotchester !

**Pull request merged**